packages/next/README.md

```diff
@@ -1878,6 +1879,99 @@ Next.js supports IE11 and all modern browsers out of the box using [`@babel/pres
+    - Pages have a user-accessible (optimized) version of the page and a search-engine indexable (unoptimized) version of the page
+    - Opt-in via `withAmp(Component)`
+- Hybrid
+    - Pages are able to be rendered as traditional HTML (default) and AMP HTML (by adding `?amp=1` to the URL)
+    - The AMP version of the page is not optimized with AMP Optimizer so that it is indexable by search-engines
+    - Opt-in via `withAmp(Component, { hybrid: true })`
+
+Both of these page modes provide a consistently fast experience for users accessing pages through search engines.
+
+### AMP Behavior with `next export`
+
+When using `next export` to statically pre-render pages Next.js will detect if the page supports AMP and change the exporting behavior based on that.
+
+For hybrid AMP `pages/about.js` would hold:
+
+- `out/about/index.html` - with client-side React runtime
+- `out/about.amp/index.html` -  AMP page
+
+For full AMP pages:
+
+- `out/about/index.html` - Optimized AMP page
+- `out/about.amp/index.html` -  AMP page
+
+During export we will automatically detect if a page is AMP only and apply these dirty/clean optimizations. The dirty version will be output to `page/index.html` and the clean version will be output to `page.amp/index.html`. We also automatically take care of adding the `<link rel="amphtml" href="/page.amp" />` and `<link rel="canonical" href="/" />` tags for you. 
+
+### Adding AMP Components
+
+To make AMP pages more interactive, the AMP community provides a bunch of components ([see some here](https://amp.dev/documentation/components/)). To add one of these components you add the script pointing to the component's code using `next/head` and then use it. See example below for more info.
+
+```js
+// pages/hello.js
+import Head from 'next/head'
+import { withAmp } from 'next/amp'
+
+export default withAmp(function MyAmpPage() {
+  return (
+    <div>
+      <Head>
+        <script
+          async
+          key="amp-timeago"
+          custom-element="amp-timeago"
+          src="https://cdn.ampproject.org/v0/amp-timeago-0.1.js"
+        />
+      </Head>
+
+      <p>Some time: {date.toJSON()}</p>
+      <amp-timeago
+        width="0"
+        height="15"
+        datetime={date.toJSON()}
+        layout="responsive"
+      >
+        .
+      </amp-timeago>
+    </div>
+  )
+})
+```
+
+### AMP Validating
+
+To help make sure only valid AMP pages are output we automatically test AMP pages with [amphtml-validator](https://www.npmjs.com/package/amphtml-validator). In development mode, any errors or warnings from validation will appear in the terminal where you started Next.js. During an export if there are validation errors, they will all be logged to the terminal at the end of the export and it will exit with status code 1 since it is not a valid export.
```

Comment:
```suggestion
AMP pages are automatically validated with [amphtml-validator](https://www.npmjs.com/package/amphtml-validator) during development. Errors and warnings will appear in the terminal where you started Next.js.

Pages are also validated during `next export` and any warnings / errors will be printed to the terminal.
Any AMP errors will cause `next export` to exit with status code `1` because the export is not valid AMP.
```